mlflow/store/artifact/hdfs_artifact_repo.py

```diff
@@ -33,8 +33,7 @@ def log_artifact(self, local_file, artifact_path=None):
         with hdfs_system(scheme=self.scheme, host=self.host, port=self.port) as hdfs:
             _, file_name = os.path.split(local_file)
             destination = posixpath.join(hdfs_base_path, file_name)
-            with hdfs.open(destination, "wb") as output:
-                output.write(open(local_file, "rb").read())
+            hdfs.upload(destination, open(local_file, "rb"))
```

Comment:
Nit: Can we use `open()` as a context manager to make sure that the file is closed after read? Can we also make that change to the other line?

```suggestion
            with open(local_file, "rb") as f:
                hdfs.upload(destination, f)
```

Reply:
Thanks for you review. Very good advice ! I  just added it.

I think the last [failure](https://github.com/mlflow/mlflow/runs/6277899802?check_suite_focus=true) might be due to the fact that it is not closed correctly. I don't have such failure when I run the unit tests on Linux though.

Reply:
Awesome! Looks like that addressed it :)